source/simulator/assembly.cc

```diff
@@ -1521,7 +1521,7 @@ namespace aspect
       }
     else
       {
-        compute_material_model_input_values (old_solution,
+        compute_material_model_input_values (current_linearization_point,
```

Comment:
Okay, I moved them before the if now. 

All I wanted to say is that in my next pull request I want to remove the whole if-else-branches, because we will probably have to do the same things for the temperature and composition anyway, I just haven't had time to test that yet. 
So I just thought I could get out of doing another commit :-)